src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,214 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+
+    auto contractDataEntry =
+        LedgerTestUtils::generateValidLedgerEntryOfType(CONTRACT_DATA);
+    contractDataEntry.lastModifiedLedgerSeq = 1;
+    ltx.createWithoutLoading(contractDataEntry);
+
+    auto classicEntry = LedgerTestUtils::generateValidLedgerEntryOfType(OFFER);
+    classicEntry.lastModifiedLedgerSeq = 1;
+    ltx.createWithoutLoading(classicEntry);
+
+    LedgerEntry TTLEntry;
+    TTLEntry.data.type(TTL);
+    TTLEntry.data.ttl().keyHash = getTTLKey(contractDataEntry).ttl().keyHash;
+    TTLEntry.data.ttl().liveUntilLedgerSeq =
+        contractDataEntry.lastModifiedLedgerSeq + 1;
+
+    auto deadEntry =
+        LedgerTestUtils::generateValidLedgerEntryOfType(CONTRACT_DATA);
+    auto deadKey = LedgerEntryKey(deadEntry);
+    ltx.eraseWithoutLoading(deadKey);
+
+    // Insert all entries into the database.
+    std::vector<LedgerEntry> ledgerVect{classicEntry, contractDataEntry,
+                                        TTLEntry};
+    std::vector<LedgerKey> deadKeyVect{deadKey};
+    app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect, deadKeyVect);
+    ltx.commit();
+
+    auto addTxn = [&](bool enoughQuota, std::vector<LedgerEntry> entries,
+                      std::vector<LedgerKey> deadKeys = {}) {
+        SorobanResources resources;
+
+        for (auto const& e : entries)
+        {
+            auto k = LedgerEntryKey(e);
+            keysToPrefetch.emplace(k);
+            if (k.type() != TTL)
+            {
+                resources.readBytes += xdr::xdr_size(e);
+                // Randomly add the key to either the read or write set.
+                if (stellar::rand_flip())
+                {
+                    resources.footprint.readOnly.emplace_back(k);
+                }
+                else
+                {
+                    resources.footprint.readWrite.emplace_back(k);
+                }
+            }
+        }
+        if (!enoughQuota)
+        {
+            resources.readBytes -= 1;
+        }
+
+        for (auto& k : deadKeys)
+        {
+            resources.footprint.readOnly.emplace_back(k);
+        }
+
+        lkMeterHot->addTxn(resources);
+        lkMeterCold->addTxn(resources);
+    };
+
+    auto checkPrefetch = [&](std::set<LedgerKey> const& expectedSuccessKeys) {
+        LedgerTxn ltx2(root);
+        auto numLoadedCold = root.prefetchSoroban(keysToPrefetch, lkMeterCold);
+        REQUIRE(numLoadedCold == expectedSuccessKeys.size());
+
+        auto preLoadPrefetchHitRate = root.getPrefetchHitRate();
+        REQUIRE(preLoadPrefetchHitRate == 0);
+        for (auto const& k : expectedSuccessKeys)
+        {
+            ltx2.load(k);
+        }
+
+        auto numLoadedHot = root.prefetchSoroban(keysToPrefetch, lkMeterHot);
+        REQUIRE(numLoadedHot == 0);
+        // 100% hit rate but make it floating point
+        REQUIRE(fabs(ltx2.getPrefetchHitRate() - 1.0f) < 0.0001f);
+    };
+
+    std::vector<LedgerEntry> allEntries;
```

Comment:
`allEntries` is unused and can be removed (you can just use `tx1Entries` to populate `expectedSuccessKeys`).